src/cmakeTaskProvider.ts

```diff
@@ -65,45 +68,66 @@ export class CMakeTask extends vscode.Task {
 export class CMakeTaskProvider implements vscode.TaskProvider {
     static CMakeScriptType: string = 'cmake';
     static CMakeSourceStr: string = "CMake";
-    private cmakeDriver?: CMakeDriver;
-    private defaultTargets: string[] = [allTargetName];
+    //private cmakeTools?: CMakeTools;
+    //private defaultTargets?: string[];
 
     constructor() {
     }
 
-    public updateCMakeDriver(cmakeDriver: CMakeDriver) {
-        this.cmakeDriver = cmakeDriver;
-        allTargetName = this.cmakeDriver.allTargetName;
-    }
-
-    public updateDefaultTargets(defaultTargets?: string[]) {
-        this.defaultTargets = (defaultTargets && defaultTargets.length > 0) ? defaultTargets :
-            this.cmakeDriver ? [this.cmakeDriver.allTargetName] : [allTargetName];
-    }
-
     public async provideTasks(): Promise<CMakeTask[]> {
         const result: CMakeTask[] = [];
-        result.push(await this.provideTask(CommandType.build));
-        result.push(await this.provideTask(CommandType.config));
-        result.push(await this.provideTask(CommandType.install));
-        result.push(await this.provideTask(CommandType.test));
-        result.push(await this.provideTask(CommandType.clean));
-        result.push(await this.provideTask(CommandType.cleanRebuild));
+        const cmakeTools: CMakeTools | undefined = getCMakeToolsForActiveFolder();
+        const cmakeDriver:  CMakeDriver | undefined = (await cmakeTools?.getCMakeDriverInstance()) || undefined;
+        result.push(await this.provideTask(CommandType.build, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.config, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.install, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.test, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.clean, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.cleanRebuild, cmakeDriver, cmakeTools?.useCMakePresets));
         return result;
     }
 
-    public async provideTask(commandType: CommandType): Promise<CMakeTask> {
+    public async provideTask(commandType: CommandType, cmakeDriver?: CMakeDriver, useCMakePresets?: boolean): Promise<CMakeTask> {
         const taskName: string = localizeCommandType(commandType);
+        let targets: string[] | undefined;
+        let preset: string | undefined;
+        const options: { cwd?: string ; environment?: Environment } = {};
+        if (!useCMakePresets) {
+            if (commandType === CommandType.build) {
+                if (cmakeDriver?.targetsName) {
```

Comment:
Do we really want to list all the targets by default?  There could be hundreds!

Reply:
I replaced this with default targets.